client/mysql.cc

```diff
@@ -1337,26 +1337,45 @@ int main(int argc,char *argv[])
   initialize_readline();
   if (!status.batch && !quick && !opt_html && !opt_xml)
   {
-    /* read-history from file, default ~/.mysql_history*/
-    if (getenv("MYSQL_HISTFILE"))
-      histfile=my_strdup(PSI_NOT_INSTRUMENTED, getenv("MYSQL_HISTFILE"),MYF(MY_WME));
-    else if (getenv("HOME"))
+    const char *home;
+    /* read-history from file, default ~/.mariadb_history*/
+    if ((histfile= getenv("MARIADB_HISTFILE"))
+        || (histfile= getenv("MYSQL_HISTFILE")))
+      histfile=my_strdup(PSI_NOT_INSTRUMENTED, histfile, MYF(MY_WME));
+    else if ((home= getenv("HOME")))
     {
       histfile=(char*) my_malloc(PSI_NOT_INSTRUMENTED,
-            strlen(getenv("HOME")) + strlen("/.mysql_history")+2, MYF(MY_WME));
+            strlen(home) + strlen("/.mariadb_history")+2, MYF(MY_WME));
       if (histfile)
-	sprintf(histfile,"%s/.mysql_history",getenv("HOME"));
+	sprintf(histfile,"%s/.mariadb_history", home);
+      if (my_access(histfile, F_OK))
+      {
+	/* no .mariadb_histroy, look for historical name and use if present */
+        char *m_histfile=(char*) my_malloc(PSI_NOT_INSTRUMENTED,
+            strlen(home) + strlen("/.mysql_history")+2, MYF(MY_WME));
```

Comment:
why do you need a second `malloc` ? you can reuse `histfile`, can you not?

Reply:
restructured not to use second malloc, just repopulated with .mariadb_history after mysql history wasn't there.